File: solutions/3501-delete-nodes-from-linked-list-present-in-array/solution.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {number[]} nums
 * @param {ListNode} head
 * @return {ListNode}
 */
var modifiedList = function(nums, head) {
    const s = new Set(nums)


    let node = head;
    while(node.next){
        if(s.has(node.next.val)){
            node.next = node.next.next

        }else{
            node = node.next
        }
    }
    if(s.has(head.val)){
        head = head.next
    }
    return head
};

File: solutions/3501-delete-nodes-from-linked-list-present-in-array/solution.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def modifiedList(self, nums: List[int], head: Optional[ListNode]) -> Optional[ListNode]:
        temp = ListNode(0, head)
        s = set(nums)

        node = temp
        while node:
            next = node.next
            while next and next.val in s:
                next = next.next
            node.next = next

            node = node.next
            
        return temp.next
            
